akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -85,7 +85,7 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
     connections.map {
       case Tcp.IncomingConnection(localAddress, remoteAddress, flow) ⇒
         val layer = serverLayer(settings, Some(remoteAddress), log)
-        val flowWithTimeoutRecovered = flow.recover { case t: TimeoutException ⇒ throw new HttpConnectionTimeoutException(t.getMessage) }
+        val flowWithTimeoutRecovered = flow.via(MapError { case t: TimeoutException ⇒ new HttpConnectionTimeoutException(t.getMessage) })
```

Comment:
Ah! Thanks to @drewhk I see why that's the fix. So if we throw in a stage we log, and if we invoke failure we do not log automagically, and then the proper debug/error logging applies.

Thanks a lot @jrudolph for the fix!